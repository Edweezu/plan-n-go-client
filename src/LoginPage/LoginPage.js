import React, { Component } from 'react'
import LoginForm from '../LoginForm/LoginForm'


export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from ||'/dashboard'
    console.log('destinationnn', destination)
    history.push(destination)
  }

  render() {
    return (
      <main className='LoginPage'>
        <h2>Login</h2>
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
      </main>
    )
  }
}