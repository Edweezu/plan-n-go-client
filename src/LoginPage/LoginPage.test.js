import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import LoginPage from './LoginPage'


describe('LoginPage component', () => {
    it('renders the given component', () => {
        const wrapper = shallow(<LoginPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})