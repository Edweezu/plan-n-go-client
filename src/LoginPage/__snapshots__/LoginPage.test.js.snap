// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders the given component 1`] = `
<main
  className="LoginPage"
>
  <h2>
    Login
  </h2>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</main>
`;
